lib/internal/modules/esm/hooks.js

```diff
@@ -789,11 +782,9 @@ function pluckHooks({
 function nextHookFactory(current, meta, { validateArgs, validateOutput }) {
   // First, prepare the current
   const { hookName } = meta;
-  const {
-    fn: hook,
-    url: hookFilePath,
-    next,
-  } = current;
```

Comment:
Did anything change here? It looks like it's now just less readable

Reply:
Yes. The previous structure of chain:

```ts
type Fn = (arg0: any, context: any, next: Fn) => any;

interface ChainItem {
  fn: Fn
  url: string;
}

type Chain = ChainItem[];
```

The new structure of chain:

```ts
type Fn = (arg0: any, context: any, next: Fn) => any;

interface LoaderInstance {
  load?: Fn;
  resolve?: Fn;
  initialize?: Fn;
  url: string;
}

interface ChainItem {
  fn: Fn
  loader: LoaderInstance;
}

type Chain = ChainItem[];
```